### PR TITLE
deprecate snapshot backups

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: definitions/scenarios/backup.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ def compose
       when :offline
         add_offline_backup_steps
       when :snapshot
+        warn!("Snapshot backups are deprecated and will be removed in a future version.")
```
Comment (ehelms) [inline definitions/scenarios/backup.rb:36]:
```suggestion
        warn("WARNING: Snapshot backups are deprecated and will be removed in a future version.")
```

**Comment by ehelms** [comment]
What about adding a DEPRECATED to the description -- https://github.com/theforeman/foreman_maintain/blob/master/lib/foreman_maintain/cli/backup_command.rb#L158?

**Comment by evgeni** [comment]
Mhh. This is not really visible:
```console
[root@centos8-stream-foreman-nightly ~]# foreman-maintain backup snapshot /lol
WARNING: Snapshot backups are deprecated and will be removed in a future version.
Starting backup: 2023-10-09 08:10:25 +0000
Running preparation steps required to run the next scenarios
================================================================================
Make sure Foreman DB is up: 
/ Checking connection to the Foreman DB                               [OK]      
--------------------------------------------------------------------------------
```

**Comment by evgeni** [comment]
now:

```console
# foreman-maintain backup snapshot /lol
Starting backup: 2023-10-09 08:46:00 +0000
Running preparation steps required to run the next scenarios
================================================================================
Make sure Foreman DB is up: 
/ Checking connection to the Foreman DB                               [OK]      
--------------------------------------------------------------------------------


Running Backup
================================================================================
Snapshot backups are deprecated:                                      [WARNING]
Snapshot backups are deprecated and will be removed in a future version.
--------------------------------------------------------------------------------
Confirm turning off services is allowed: 
WARNING: This script will stop your services.
```

File: definitions/scenarios/backup.rb [hunk 2]
```diff
@@ -21,6 +21,7 @@ class Backup < ForemanMaintain::Scenario
 
     def compose
       check_valid_startegy
```
Comment (ehelms) [inline definitions/scenarios/backup.rb:23]: heh -- just noticed this fun typo `startegy`

**Review by ehelms — APPROVED** [review]

**Pull request merged**